memgpt/data_types.py

```diff
@@ -146,3 +147,146 @@ def __init__(
+        self.default_model = default_model
+        self.default_model_endpoint_type = default_model_endpoint_type
+        self.default_model_endpoint = default_model_endpoint
+        self.default_model_wrapper = default_model_wrapper
+        if default_context_window is None:
+            self.default_context_window = (
+                LLM_MAX_TOKENS[self.default_model] if self.default_model in LLM_MAX_TOKENS else LLM_MAX_TOKENS["DEFAULT"]
+            )
+        else:
+            self.default_context_window = default_context_window
+
+        # defaults: embeddings
+        self.default_embedding_endpoint_type = default_embedding_endpoint_type
+        self.default_embedding_endpoint = default_embedding_endpoint
+        self.default_embedding_model = default_embedding_model
+        self.default_embedding_dim = default_embedding_dim
+        self.default_embedding_chunk_size = default_embedding_chunk_size
+
+        # azure information
+        self.azure_key = azure_key
+        self.azure_endpoint = azure_endpoint
+        self.azure_version = azure_version
+        self.azure_deployment = azure_deployment
+
+        # openai information
+        self.openai_key = openai_key
+
+        # misc
+        self.memgpt_version = memgpt_version
+
+        # TODO: generate
+        self.anon_clientid = anon_clientid
+        self.policies_accepted = policies_accepted
+
+
+class AgentState(Record):
+    def __init__(
+        self,
+        name: str,
+        persona_file: str,  # the filename where the persona was originally sourced from
+        human_file: str,  # the filename where the human was originally sourced from
+        # (in-context) state contains:
+        # persona: str  # the current persona text
+        # human: str  # the current human text
+        # system: str,  # system prompt (not required if initializing with a preset)
+        # functions: dict,  # schema definitions ONLY (function code linked at runtime)
+        # messages: List[dict],  # in-context messages
+        state: Optional[dict] = None,
+        # model info
+        model: Optional[str] = None,
+        model_endpoint_type: Optional[str] = None,
+        model_endpoint: Optional[str] = None,
+        model_wrapper: Optional[str] = None,
+        context_window: Optional[int] = None,  # TODO(swooders) I don't think this should be optional
+        # embedding info
+        embedding_endpoint_type: Optional[str] = None,
+        embedding_endpoint: Optional[str] = None,
+        embedding_model: Optional[str] = None,
+        embedding_dim: Optional[int] = None,  # TODO(swooders) similarly these probably should not be optional
+        embedding_chunk_size: Optional[int] = None,
+        # other
+        preset: Optional[str] = None,
+        data_sources: Optional[list] = None,
+        create_time: Optional[str] = None,
+        memgpt_version: Optional[str] = None,
+    ):
+        # TODO(swooders) we need to handle the case where name is None here
```

Comment:
TODO how should we handle defaults here? exact same as agentconfig?

Reply:
What do you think of not allowing any defaults, to enforce that everything is passed in from the `User` record (which defines all the default values)?